controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -914,35 +601,117 @@ func createTrainingJob(k8sName, sageMakerName string, hasFinalizer bool) *traini
 			StoppingCondition: &commonv1.StoppingCondition{},
 		},
 		Status: trainingjobv1.TrainingJobStatus{
-			TrainingJobStatus: InitializingJobStatus,
+			TrainingJobStatus: controllers.InitializingJobStatus,
 		},
 	}
 }
 
-// Helper function to create a sagemaker.DescribeTrainingJobOutput from an TrainingJob.
-func createDescriptionFromSmTrainingJob(job *trainingjobv1.TrainingJob) sagemaker.DescribeTrainingJobOutput {
-	return sagemaker.DescribeTrainingJobOutput{
-		TrainingJobName: job.Spec.TrainingJobName,
-		AlgorithmSpecification: &sagemaker.AlgorithmSpecification{
-			TrainingInputMode: sagemaker.TrainingInputMode(job.Spec.AlgorithmSpecification.TrainingInputMode),
-		},
-		OutputDataConfig: &sagemaker.OutputDataConfig{
-			S3OutputPath: job.Spec.OutputDataConfig.S3OutputPath,
-		},
-		ResourceConfig: &sagemaker.ResourceConfig{
-			InstanceCount:  job.Spec.ResourceConfig.InstanceCount,
-			InstanceType:   sagemaker.TrainingInstanceType(job.Spec.ResourceConfig.InstanceType),
-			VolumeSizeInGB: job.Spec.ResourceConfig.VolumeSizeInGB,
-		},
-		RoleArn:           job.Spec.RoleArn,
-		StoppingCondition: &sagemaker.StoppingCondition{},
-	}
+// Add a finalizer to the deployment.
+func AddFinalizer(trainingJob *trainingjobv1.TrainingJob) {
+	var actual trainingjobv1.TrainingJob
+	err := k8sClient.Get(context.Background(), types.NamespacedName{
+		Namespace: trainingJob.ObjectMeta.Namespace,
+		Name:      trainingJob.ObjectMeta.Name,
+	}, &actual)
+	Expect(err).ToNot(HaveOccurred())
+
+	actual.ObjectMeta.Finalizers = []string{controllers.SageMakerResourceFinalizerName}
+
+	Expect(k8sClient.Update(context.Background(), &actual)).To(Succeed())
+}
+
+// Set the deletion timestamp to be nonzero.
+func SetDeletionTimestamp(trainingJob *trainingjobv1.TrainingJob) {
+	err := k8sClient.Get(context.Background(), types.NamespacedName{
+		Namespace: trainingJob.ObjectMeta.Namespace,
+		Name:      trainingJob.ObjectMeta.Name,
+	}, trainingJob)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(k8sClient.Delete(context.Background(), trainingJob)).To(Succeed())
+}
+
+// Expect the controller return value to be RequeueAfterInterval, with the poll duration specified.
+func ExpectRequeueAfterInterval(result ctrl.Result, err error, pollDuration string) {
```

Comment:
I see some of these are common to `HostingDeployment` controller tests. Can we move the generic helpers to the [common testing package](https://github.com/aws/amazon-sagemaker-operator-for-k8s/tree/master/controllers/controllertest) or subpackage?